### DISCUSSION TITLE
Update for new promises in mirai 1.0.0

### DISCUSSION THREAD

**Discussion by shikokuchuo** [hypebright/async_shiny#13] [General]
Hi @hypebright someone pointed this repo out to me the other day. It's fantastic that you seem to have been a very early adopter of `mirai`!

I am not sure if you've followed the latest re. Shiny ExtendedTask / mirai integration, but the new 'mirai' promises are completely event-driven without the polling `later()` loop of other types of promise, and represent the next level in terms of responsiveness and scalability. These were developed in collaboration with Joe Cheng.

You may refer to this vignette here: https://shikokuchuo.net/mirai/articles/shiny.html

If you have any questions, do feel free to reach out.

**Comment by hypebright** [comment]
Hey @shikokuchuo, excellent idea! I indeed saw some work being done on `mirai` and it looks very promising. Will definitely look at it and try to add an example when I have a bit more time on my hands. In the meantime, feel free to open a PR with a new example in the mirai folder that demonstrates this new functionality :).

**Reply by shikokuchuo** [reply]
I've opened #14 to correct the description as `mirai.promises` is now archived and some obvious improvements to the existing example. Please feel free to take / modify any `ExtendedTask` example from my vignette, or it should be straightforward to create some new ones. Also Joe Cheng based his ExtendedTask examples on your stock ticker app (and he gives credit to you) - I'm sure you've seen at https://rpubs.com/jcheng/beyond-async

**Reply by hypebright** [reply]
Yes I saw that at the ShinyConf, awesome! Thanks for the PR 👍

**Reply by hypebright** [reply]
Just added an example with `ExtendedTask` too, nice!

**Reply by shikokuchuo** [reply]
Fantastic! Love the 'running' icon you put in - that's a nice touch! I hope this means that you found the `mirai` interface intuitive enough - you seem to have picked it up straight away.

I'd love to go through the underlying way these 'next gen' promises work, but you might have already figured out... in any case, feel free to drop me a message separately.

**Comment by hypebright** [comment]
Closing this since it's integrated
